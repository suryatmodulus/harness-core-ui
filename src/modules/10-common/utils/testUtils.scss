/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.browser {
  border: 1px solid #cccccc;
  border-radius: 8px;
  min-height: 300px;
  overflow: auto;

  .browser-header {
    background-color: #cccccc;
    padding: 8px 16px;
    display: flex;
  }

  .browser-path {
    display: block;
    border: none;
    line-height: 14px;
    font-size: 12px;
    color: #444444;
    border-radius: 9px;
    min-width: 100px;
    padding: 2px 8px;
    flex: 1 1 auto;
    border: 1px solid transparent;

    &:focus {
      border-color: rgb(30, 167, 253);
      outline: none;
    }
  }

  .browser-content {
    padding: 8px;
  }
}
