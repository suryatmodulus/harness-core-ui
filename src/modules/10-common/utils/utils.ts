/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import type { IconName } from '@harness/uicore'
import { Scope } from '@common/interfaces/SecretsInterface'
import { ModuleName } from 'framework/types/ModuleName'

interface SetPageNumberProps {
  setPage: (value: React.SetStateAction<number>) => void
  pageItemsCount?: number
  page: number
}

export const getModuleIcon = (module: ModuleName): IconName => {
  switch (module) {
    case ModuleName.CD:
      return 'cd-main'
    case ModuleName.CV:
      return 'cv-main'
    case ModuleName.CI:
      return 'ci-main'
    case ModuleName.CE:
      return 'ce-main'
    case ModuleName.CF:
      return 'cf-main'
  }
  return 'nav-project'
}

export const getReference = (scope?: Scope, identifier?: string): string | undefined => {
  switch (scope) {
    case Scope.PROJECT:
      return identifier
    case Scope.ORG:
      return `org.${identifier}`
    case Scope.ACCOUNT:
      return `account.${identifier}`
  }
}

export const setPageNumber = ({ setPage, pageItemsCount, page }: SetPageNumberProps): void => {
  if (pageItemsCount === 0 && page > 0) {
    setPage(page - 1)
  }
}

export const formatCount = (num: number): string => {
  const min = 1e3
  if (num >= min) {
    const units = ['k', 'M', 'B', 'T']
    const order = Math.floor(Math.log(num) / Math.log(1000))
    const finalNum = Math.round(num / 1000 ** order)
    if (finalNum === min && order < units.length) {
      return 1 + units[order]
    }
    return finalNum + units[order - 1]
  }
  return num.toLocaleString()
}
