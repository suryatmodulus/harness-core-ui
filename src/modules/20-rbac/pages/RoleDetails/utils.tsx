/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Free Trial 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/05/PolyForm-Free-Trial-1.0.0.txt.
 */

import produce from 'immer'
import { PermissionIdentifier } from '@rbac/interfaces/PermissionIdentifier'
import type { ResourceType } from '@rbac/interfaces/ResourceType'
import type { Permission, PermissionResponse } from 'services/rbac'

export const getPermissionMap = (permissionList?: PermissionResponse[]): Map<ResourceType, Permission[]> => {
  const permissionsMap: Map<ResourceType, Permission[]> = new Map()
  permissionList?.map(response => {
    if (response.permission.resourceType) {
      if (permissionsMap.has(response.permission.resourceType as ResourceType)) {
        permissionsMap.get(response.permission.resourceType as ResourceType)?.push(response.permission)
      } else permissionsMap.set(response.permission.resourceType as ResourceType, [response.permission])
    }
  })
  return permissionsMap
}

export const onPermissionChange = (
  permission: string,
  isAdd: boolean,
  permissions: string[],
  setPermissions: (value: React.SetStateAction<string[]>) => void
): void => {
  if (isAdd) {
    if (
      permission === PermissionIdentifier.EDIT_DASHBOARD ||
      permissions.indexOf(PermissionIdentifier.EDIT_DASHBOARD) !== -1
    ) {
      setPermissions(_permissions => [...permissions, permission, PermissionIdentifier.VIEW_DASHBOARD])
    } else {
      setPermissions(_permissions => [...permissions, permission])
    }
  } else if (
    !(
      permission === PermissionIdentifier.VIEW_DASHBOARD &&
      permissions.indexOf(PermissionIdentifier.EDIT_DASHBOARD) !== -1
    )
  ) {
    setPermissions(_permissions =>
      produce(_permissions, draft => {
        draft?.splice(permissions.indexOf(permission), 1)
      })
    )
  }
}
