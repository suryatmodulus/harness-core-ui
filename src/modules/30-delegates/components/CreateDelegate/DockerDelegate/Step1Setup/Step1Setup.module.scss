/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.baseContainer {
  height: 500px;
  display: flex;
}

.leftPanel {
  overflow-y: auto;
  flex: 1 !important;
}

.rightPanel {
  flex: 1;
}

.delegateForm {
  padding: var(--spacing-xsmall) !important;
  min-height: 580px;
  overflow: auto;
}

.formGroup {
  display: grid;
  grid-template-columns: 1fr 30px;
  align-items: center;
}

.verificationField {
  box-shadow: 0px 0px 5px rgba(40, 41, 61, 0.0348285);
  background: #effbff !important;
  border-radius: 4px !important;
}

.tokensSelectContainer {
  align-items: start;
  button {
    margin-top: 21px !important;
  }
}
