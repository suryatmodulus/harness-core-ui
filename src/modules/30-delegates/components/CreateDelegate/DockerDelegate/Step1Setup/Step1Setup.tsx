/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React, { useState, useMemo } from 'react'
import { useParams } from 'react-router-dom'
import type { FormikActions } from 'formik'
import set from 'lodash-es/set'
import {
  Layout,
  Formik,
  Button,
  FormikForm,
  Container,
  StepProps,
  SelectOption,
  FormInput
} from '@wings-software/uicore'
import * as Yup from 'yup'
import { validateDockerDelegatePromise, useGetDelegateTokens } from 'services/portal'
import type { DelegateTokenDetails } from 'services/portal'

import { useStrings } from 'framework/strings'

import type { ProjectPathProps } from '@common/interfaces/RouteInterfaces'
import { AddDescriptionAndKVTagsWithIdentifier } from '@common/components/AddDescriptionAndTags/AddDescriptionAndTags'
import { useCreateTokenModal } from '@delegates/components/DelegateTokens/modals/useCreateTokenModal'

import type { DockerDelegateWizardData } from '../CreateDockerDelegate'
import css from './Step1Setup.module.scss'

interface DelegateSetupStepProps {
  onBack?: any
}

//this regex is retrieved from kubernetes
const delegateNameRegex = /^[a-z]([-a-z0-9]*[a-z])?(\.[a-z0-9]([-a-z0-9]*[a-z])?)*$/g

const formatTokenOptions = (data: any): Array<SelectOption> => {
  const tokens: Array<DelegateTokenDetails> = data?.resource

  return tokens
    ? tokens.map((item: DelegateTokenDetails) => {
        return { label: item.name || '', value: item.name || '' }
      })
    : []
}

const Step1Setup: React.FC<StepProps<DockerDelegateWizardData> & DelegateSetupStepProps> = props => {
  const { prevStepData } = props
  let initialValues
  if (prevStepData) {
    const tags = {}
    prevStepData.tags?.forEach(tag => set(tags, tag, ''))
    initialValues = {
      ...prevStepData,
      tags: tags
    }
  } else {
    initialValues = {
      name: '',
      identifier: '',
      description: '',
      tags: {},
      tokenName: ''
    }
  }

  const { accountId, projectIdentifier, orgIdentifier } = useParams<ProjectPathProps>()
  const { getString } = useStrings()

  const [formData, setInitValues] = useState<DockerDelegateWizardData>(initialValues as DockerDelegateWizardData)

  const validateName = async (
    values: DockerDelegateWizardData,
    formikActions: FormikActions<DockerDelegateWizardData>
  ) => {
    const response = (await validateDockerDelegatePromise({
      queryParams: {
        accountId,
        projectIdentifier,
        orgIdentifier,
        delegateName: values.name,
        tokenName: values.tokenName
      }
    })) as any
    const isNameUnique = !response?.responseMessages[0]

    if (isNameUnique) {
      const stepPrevData = {
        name: values.name,
        identifier: values.identifier,
        description: values.description,
        tokenName: values.tokenName
      }
      const tagsArray = Object.keys(values.tags || {})
      set(stepPrevData, 'tags', tagsArray)
      props?.nextStep?.(stepPrevData)
    } else {
      formikActions.setFieldError('name', getString('delegates.delegateNameNotUnique'))
    }
  }

  const { data: tokensResponse, refetch: getTokens } = useGetDelegateTokens({
    queryParams: {
      accountId,
      projectIdentifier,
      orgIdentifier,
      status: 'ACTIVE'
    }
  })
  const defaultToken = tokensResponse?.resource?.[0]

  const { openCreateTokenModal } = useCreateTokenModal({ onSuccess: getTokens })

  React.useEffect(() => {
    if (defaultToken) {
      formData.tokenName = defaultToken?.name
      setInitValues({ ...formData })
    }
  }, [defaultToken])

  const onSubmit = (values: DockerDelegateWizardData, formikActions: FormikActions<DockerDelegateWizardData>) => {
    setInitValues(values)
    validateName(values, formikActions)
  }

  const delegateTokenOptions = useMemo(() => formatTokenOptions(tokensResponse), [tokensResponse])

  return (
    <Layout.Vertical padding="xxlarge">
      <Container padding="small">
        <Formik
          initialValues={formData}
          onSubmit={onSubmit}
          formName="delegateSetupStepForm"
          validationSchema={Yup.object().shape({
            name: Yup.string()
              .trim()
              .required(getString('delegate.delegateNameRequired'))
              .matches(delegateNameRegex, getString('delegates.delegateNameRegexIssue')),
            tokenName: Yup.string().required(getString('delegates.tokens.tokenRequired'))
          })}
        >
          {() => {
            return (
              <FormikForm>
                <Container className={css.delegateForm}>
                  <Layout.Horizontal className={css.baseContainer}>
                    <Layout.Vertical className={css.leftPanel}>
                      <div className={css.formGroup}>
                        <AddDescriptionAndKVTagsWithIdentifier
                          forceOpenTags
                          identifierProps={{
                            inputLabel: getString('delegate.delegateName')
                          }}
                        />
                      </div>
                      <Layout.Horizontal className={css.tokensSelectContainer} spacing="small">
                        <FormInput.Select
                          items={delegateTokenOptions}
                          label={getString('delegates.tokens.delegateTokens')}
                          name="tokenName"
                        />
                        <Button
                          minimal
                          icon="plus"
                          onClick={e => {
                            e.preventDefault()
                            openCreateTokenModal()
                          }}
                          text={getString('add')}
                        />
                      </Layout.Horizontal>
                    </Layout.Vertical>
                    <Layout.Vertical className={css.rightPanel} />
                  </Layout.Horizontal>
                </Container>
                <Layout.Horizontal>
                  <Button
                    id="delegateSetupBackBtn"
                    intent="none"
                    text={getString('back')}
                    onClick={props.onBack}
                    icon="chevron-left"
                    margin={{ right: 'small' }}
                  />
                  <Button type="submit" intent="primary" text={getString('continue')} rightIcon="chevron-right" />
                </Layout.Horizontal>
              </FormikForm>
            )
          }}
        </Formik>
      </Container>
    </Layout.Vertical>
  )
}

export default Step1Setup
