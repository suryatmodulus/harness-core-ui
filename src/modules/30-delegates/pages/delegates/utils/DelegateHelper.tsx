/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { useStrings } from 'framework/strings'
import { DelegateTypes } from '@delegates/constants'

export const GetDelegateTitleTextByType = (type: string): string => {
  const { getString } = useStrings()

  switch (type) {
    case DelegateTypes.KUBERNETES_CLUSTER:
      return getString('kubernetesText')
    default:
      /* istanbul ignore next */
      return ''
  }
}
