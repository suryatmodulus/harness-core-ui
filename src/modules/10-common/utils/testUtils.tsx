/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { UseGetProps, UseGetReturn, RestfulProvider } from 'restful-react'
import { queryByAttribute } from '@testing-library/react'
import { compile } from 'path-to-regexp'
import { createMemoryHistory } from 'history'
import { Router, Route, Switch, useLocation, useHistory } from 'react-router-dom'
import { ModalProvider } from '@wings-software/uicore'
import qs from 'qs'

import { enableMapSet } from 'immer'
import { AppStoreContext, AppStoreContextProps } from 'framework/AppStore/AppStoreContext'
import { LicenseStoreContext, LicenseStoreContextProps } from 'framework/LicenseStore/LicenseStoreContext'
import { LICENSE_STATE_VALUES } from 'framework/LicenseStore/licenseStoreUtil'
import { withAccountId, accountPathProps } from '@common/utils/routeUtils'
import type { Project } from 'services/cd-ng'
import { StringsContext } from 'framework/strings'

import { FeaturesContext, FeaturesContextProps } from 'framework/featureStore/FeaturesContext'
import type { FeatureDetail, FeatureMetaData } from 'framework/featureStore/featureStoreUtil'
import type { FeatureIdentifier } from 'framework/featureStore/FeatureIdentifier'
import './testUtils.scss'
import { PermissionsContext, PermissionsContextProps } from 'framework/rbac/PermissionsContext'

export type UseGetMockData<TData, TError = undefined, TQueryParams = undefined, TPathParams = undefined> = Required<
  UseGetProps<TData, TError, TQueryParams, TPathParams>
>['mock']

export interface UseGetMockDataWithMutateAndRefetch<T> extends UseGetMockData<T> {
  mutate: () => Record<string, unknown>
  refetch: () => Record<string, unknown>
  cancel?: () => Record<string, unknown>
}

export interface UseMutateMockData<TData, TRequestBody = unknown> {
  loading?: boolean
  mutate?: (data?: TRequestBody) => Promise<TData>
}

export type UseGetReturnData<TData, TError = undefined, TQueryParams = undefined, TPathParams = undefined> = Omit<
  UseGetReturn<TData, TError, TQueryParams, TPathParams>,
  'absolutePath' | 'cancel' | 'response'
>

export const findDialogContainer = (): HTMLElement | null => document.querySelector('.bp3-dialog')
export const findPopoverContainer = (): HTMLElement | null => document.querySelector('.bp3-popover-content')

export interface TestWrapperProps {
  path?: string
  pathParams?: Record<string, string | number>
  queryParams?: Record<string, unknown>
  defaultAppStoreValues?: Partial<AppStoreContextProps>
  defaultLicenseStoreValues?: Partial<LicenseStoreContextProps>
  defaultPermissionValues?: Partial<PermissionsContextProps>
  defaultFeaturesValues?: Partial<FeaturesContextProps>
  projects?: Project[]
  enableBrowserView?: boolean
  stringsData?: Record<string, string>
  getString?(key: string): string
}

export const prependAccountPath = (path: string): string => withAccountId(() => path)(accountPathProps)

export const CurrentLocation = (): JSX.Element => {
  const location = useLocation()
  return (
    <div>
      <h1>Not Found</h1>
      <div data-testid="location">{`${location.pathname}${
        location.search ? `?${location.search.replace(/^\?/g, '')}` : ''
      }`}</div>
    </div>
  )
}

export interface BrowserViewProps {
  enable?: boolean
  children: React.ReactNode
}

export function BrowserView(props: BrowserViewProps): React.ReactElement {
  const { enable, children } = props
  const location = useLocation()
  const history = useHistory()

  if (!enable) {
    return <>{children}</>
  }

  function handlePathChange(e: React.ChangeEvent<HTMLInputElement>): void {
    history.replace(e.currentTarget.value)
  }

  const search = location.search ? `?${location.search.replace(/^\?/, '')}` : ''

  return (
    <div className="browser">
      <div className="browser-header">
        <input className="browser-path" value={location.pathname + search} onChange={handlePathChange} />
      </div>
      <div className="browser-content">{children}</div>
    </div>
  )
}

export const TestWrapper: React.FC<TestWrapperProps> = props => {
  enableMapSet()
  const {
    path = '/',
    pathParams = {},
    defaultAppStoreValues,
    queryParams = {},
    defaultLicenseStoreValues,
    defaultPermissionValues,
    defaultFeaturesValues,
    stringsData = {},
    getString = (key: string) => key
  } = props

  const search = qs.stringify(queryParams, { addQueryPrefix: true })
  const routePath = compile(path)(pathParams) + search

  // eslint-disable-next-line react-hooks/exhaustive-deps
  const history = React.useMemo(() => createMemoryHistory({ initialEntries: [routePath] }), [])

  const defaultReturn = {
    enabled: true
  }

  /** TODO: Try fixing this later. This is causing some tests to fail */
  // React.useEffect(() => {
  //   history.replace(compile(path)(pathParams) + qs.stringify(queryParams, { addQueryPrefix: true }))
  //   // eslint-disable-next-line react-hooks/exhaustive-deps
  // }, [path, pathParams, queryParams])

  return (
    <StringsContext.Provider value={{ data: stringsData as any, getString }}>
      <AppStoreContext.Provider
        value={{
          featureFlags: {
            FEATURE_ENFORCEMENT_ENABLED: true
          },
          updateAppStore: () => void 0,
          currentUserInfo: {},
          ...defaultAppStoreValues
        }}
      >
        <LicenseStoreContext.Provider
          value={{
            versionMap: {},
            licenseInformation: {},
            CI_LICENSE_STATE: LICENSE_STATE_VALUES.ACTIVE,
            FF_LICENSE_STATE: LICENSE_STATE_VALUES.ACTIVE,
            CCM_LICENSE_STATE: LICENSE_STATE_VALUES.ACTIVE,
            CD_LICENSE_STATE: LICENSE_STATE_VALUES.ACTIVE,
            updateLicenseStore: () => void 0,
            ...defaultLicenseStoreValues
          }}
        >
          <PermissionsContext.Provider
            value={{
              permissions: new Map<string, boolean>(),
              requestPermission: () => void 0,
              checkPermission: () => true,
              cancelRequest: () => void 0,
              ...defaultPermissionValues
            }}
          >
            <FeaturesContext.Provider
              value={{
                features: new Map<FeatureIdentifier, FeatureDetail>(),
                featureMap: new Map<FeatureIdentifier, FeatureMetaData>(),
                getEdition: () => {
                  return undefined
                },
                requestFeatures: () => void 0,
                requestLimitFeature: () => void 0,
                checkFeature: () => {
                  return defaultReturn
                },
                checkLimitFeature: () => {
                  return defaultReturn
                },
                getRestrictionType: () => {
                  return undefined
                },
                ...defaultFeaturesValues
              }}
            >
              <Router history={history}>
                <ModalProvider>
                  <RestfulProvider base="/">
                    <BrowserView enable={props.enableBrowserView}>
                      <Switch>
                        <Route exact path={path}>
                          {props.children}
                        </Route>
                        <Route>
                          <CurrentLocation />
                        </Route>
                      </Switch>
                    </BrowserView>
                  </RestfulProvider>
                </ModalProvider>
              </Router>
            </FeaturesContext.Provider>
          </PermissionsContext.Provider>
        </LicenseStoreContext.Provider>
      </AppStoreContext.Provider>
    </StringsContext.Provider>
  )
}

export const queryByNameAttribute = (name: string, container: HTMLElement): HTMLElement | null =>
  queryByAttribute('name', container, name)
