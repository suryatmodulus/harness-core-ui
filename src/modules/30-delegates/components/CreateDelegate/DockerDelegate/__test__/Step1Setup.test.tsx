/*
 * Copyright 2022 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { render } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import Step1Setup from '../Step1Setup/Step1Setup'

jest.mock('services/portal', () => ({
  useCreateDelegateToken: jest.fn().mockImplementation(() => ({
    mutate: jest.fn().mockImplementation(() => undefined)
  })),
  useGetDelegateSizes: jest.fn().mockImplementation(() => {
    return {
      data: [
        {
          size: 'MEDIUM',
          label: 'medium',
          ram: '16',
          cpu: '4'
        }
      ],
      refetch: jest.fn(),
      error: null,
      loading: false
    }
  }),
  useGenerateDockerDelegate: jest.fn().mockImplementation(() => {
    return {
      mutate: jest.fn().mockImplementation(() => ({
        resource: {}
      }))
    }
  }),
  useGetDelegateTokens: jest.fn().mockImplementation(() => ({
    mutate: jest.fn().mockImplementation(() => ({
      resource: [
        {
          name: 'Token1'
        }
      ]
    }))
  }))
}))

describe('Create Docker Step1Setup', () => {
  test('render data', () => {
    const { container } = render(
      <TestWrapper>
        <Step1Setup />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })
})
