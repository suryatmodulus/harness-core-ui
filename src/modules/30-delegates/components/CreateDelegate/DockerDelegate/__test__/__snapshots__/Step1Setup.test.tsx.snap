// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Docker Step1Setup render data 1`] = `
<div>
  <div
    class="Layout--vertical StyledProps--main StyledProps--padding-xxlarge"
  >
    <div
      class="StyledProps--font StyledProps--main StyledProps--padding-small"
    >
      <div
        class="OverlaySpinner--overlaySpinner"
      >
        <form
          class="FormikForm--main"
        >
          <div>
            <div
              class="StyledProps--font StyledProps--main delegateForm"
            >
              <div
                class="Layout--horizontal StyledProps--main baseContainer"
              >
                <div
                  class="Layout--vertical StyledProps--main leftPanel"
                >
                  <div
                    class="formGroup"
                  >
                    <div
                      class="StyledProps--font StyledProps--main main"
                    >
                      <div
                        class="StyledProps--font StyledProps--main connectorFormNameWrapper"
                      >
                        <div
                          class="StyledProps--font StyledProps--main connectorFormNameElm"
                        >
                          <div
                            class="InputWithIdentifier--txtNameContainer TextInput--main"
                          >
                            <div
                              class="Layout--horizontal Layout--layout-spacing-xsmall StyledProps--main InputWithIdentifier--txtIdContainer"
                            >
                              <div
                                class="Text--withDocsTooltip"
                                data-tooltip-id="delegateSetupStepForm_identifier"
                              >
                                <p
                                  class="StyledProps--font StyledProps--main"
                                >
                                  ID
                                </p>
                              </div>
                              <p
                                class="StyledProps--font StyledProps--main"
                              >
                                :
                              </p>
                              <span
                                class="bp3-popover-wrapper"
                              >
                                <span
                                  class="bp3-popover-target"
                                >
                                  <div
                                    class="bp3-editable-text bp3-editable-text-placeholder InputWithIdentifier--idValue"
                                    tabindex="0"
                                  >
                                    <span
                                      class="bp3-editable-text-content"
                                      style="height: 1px; line-height: 1px; min-width: 0;"
                                    />
                                  </div>
                                </span>
                              </span>
                              <span
                                class="bp3-icon StyledProps--main"
                                data-icon="Edit"
                                style="vertical-align: middle; cursor: pointer;"
                              >
                                <svg
                                  height="12"
                                  viewBox="0 0 16 16"
                                  width="12"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    clip-rule="evenodd"
                                    d="M11.207.485a.5.5 0 01.354-.146 3.5 3.5 0 013.5 3.5.5.5 0 01-.146.353l-11 11a.5.5 0 01-.354.147h-3a.5.5 0 01-.5-.5v-3a.5.5 0 01.146-.354l11-11zm.553.862L1.061 12.046v2.293h2.293l10.7-10.7a2.5 2.5 0 00-2.294-2.292z"
                                    fill="currentColor"
                                    fill-rule="evenodd"
                                  />
                                  <path
                                    clip-rule="evenodd"
                                    d="M11.89 2.646a.5.5 0 010 .708L8.353 6.889a.5.5 0 01-.708-.707l3.536-3.536a.5.5 0 01.707 0z"
                                    fill="currentColor"
                                    fill-rule="evenodd"
                                    opacity="0.3"
                                  />
                                </svg>
                              </span>
                            </div>
                            <div
                              class="bp3-form-group"
                              data-id="name-0"
                            >
                              <label
                                class="bp3-label"
                                for="name"
                              >
                                <span
                                  class="Tooltip--acenter"
                                  data-tooltip-id="delegateSetupStepForm_name"
                                >
                                  delegate.delegateName
                                </span>
                                 
                                <span
                                  class="bp3-text-muted"
                                />
                              </label>
                              <div
                                class="bp3-form-content"
                              >
                                <div
                                  class="bp3-input-group"
                                >
                                  <input
                                    autocomplete="off"
                                    class="bp3-input"
                                    name="name"
                                    type="text"
                                    value=""
                                  />
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                        <div
                          class="Layout--vertical Layout--layout-spacing-xsmall StyledProps--main"
                          style="justify-content: center;"
                        >
                          <p
                            class="StyledProps--font StyledProps--main fieldToggleLabel StyledProps--color StyledProps--color-primary7"
                          >
                            + Description
                          </p>
                        </div>
                      </div>
                      <div
                        class="StyledProps--font StyledProps--main connectorFormElm"
                      >
                        <div
                          class="bp3-form-group expandedTags"
                          data-id="tags-1"
                        >
                          <label
                            class="bp3-label"
                            for="tags"
                          >
                            <span
                              class="Tooltip--acenter"
                              data-tooltip-id="delegateSetupStepForm_tags"
                            >
                              <div
                                class="StyledProps--font StyledProps--main headerRow"
                              >
                                <span
                                  class="StyledProps--font StyledProps--main StyledProps--inline"
                                >
                                  Tags
                                </span>
                                <span
                                  class="StyledProps--font StyledProps--main fieldToggleLabel StyledProps--inline StyledProps--color StyledProps--color-primary7"
                                >
                                  removeLabel
                                </span>
                              </div>
                            </span>
                             
                            <span
                              class="bp3-text-muted"
                            />
                          </label>
                          <div
                            class="bp3-form-content"
                          >
                            <div
                              class="bp3-input bp3-tag-input"
                            >
                              <div
                                class="bp3-tag-input-values"
                              >
                                <input
                                  class="bp3-input-ghost"
                                  data-mentions="kv-tag-input-tags}"
                                  placeholder="Type and press enter to create a tag"
                                  value=""
                                />
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="Layout--horizontal Layout--layout-spacing-small StyledProps--main tokensSelectContainer"
                  >
                    <div
                      class="bp3-form-group"
                      data-id="tokenName-2"
                    >
                      <label
                        class="bp3-label"
                        for="tokenName"
                      >
                        <span
                          class="Tooltip--acenter"
                          data-tooltip-id="delegateSetupStepForm_tokenName"
                        >
                          delegates.tokens.delegateTokens
                        </span>
                         
                        <span
                          class="bp3-text-muted"
                        />
                      </label>
                      <div
                        class="bp3-form-content"
                      >
                        <div
                          class="bp3-popover-wrapper Select--main bp3-fill"
                        >
                          <div
                            class="bp3-popover-target"
                          >
                            <div
                              class="bp3-input-group"
                            >
                              <input
                                autocomplete="off"
                                class="bp3-input"
                                name="tokenName"
                                placeholder="- Select -"
                                style="padding-right: 0px;"
                                type="text"
                                value=""
                              />
                              <span
                                class="bp3-input-action"
                              >
                                <span
                                  class="bp3-icon bp3-icon-chevron-down StyledProps--main StyledProps--padding-small"
                                  icon="chevron-down"
                                >
                                  <svg
                                    data-icon="chevron-down"
                                    height="14"
                                    viewBox="0 0 16 16"
                                    width="14"
                                  >
                                    <desc>
                                      chevron-down
                                    </desc>
                                    <path
                                      d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
                                      fill-rule="evenodd"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <button
                      class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main Button--with-current-color Button--without-shadow Button--withLeftIcon"
                      type="button"
                    >
                      <span
                        class="bp3-icon bp3-icon-plus StyledProps--main StyledProps--padding-right-xsmall"
                        icon="plus"
                      >
                        <svg
                          data-icon="plus"
                          height="16"
                          viewBox="0 0 16 16"
                          width="16"
                        >
                          <desc>
                            plus
                          </desc>
                          <path
                            d="M13 7H9V3c0-.55-.45-1-1-1s-1 .45-1 1v4H3c-.55 0-1 .45-1 1s.45 1 1 1h4v4c0 .55.45 1 1 1s1-.45 1-1V9h4c.55 0 1-.45 1-1s-.45-1-1-1z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                      <span
                        class="bp3-button-text"
                      >
                        add
                      </span>
                    </button>
                  </div>
                </div>
                <div
                  class="Layout--vertical StyledProps--main rightPanel"
                />
              </div>
            </div>
            <div
              class="Layout--horizontal StyledProps--main"
            >
              <button
                class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--margin-right-small Button--with-current-color Button--withLeftIcon"
                id="delegateSetupBackBtn"
                type="button"
              >
                <span
                  class="bp3-icon bp3-icon-chevron-left StyledProps--main StyledProps--padding-right-xsmall"
                  icon="chevron-left"
                >
                  <svg
                    data-icon="chevron-left"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                  >
                    <desc>
                      chevron-left
                    </desc>
                    <path
                      d="M7.41 8l3.29-3.29c.19-.18.3-.43.3-.71a1.003 1.003 0 00-1.71-.71l-4 4C5.11 7.47 5 7.72 5 8c0 .28.11.53.29.71l4 4a1.003 1.003 0 001.42-1.42L7.41 8z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
                <span
                  class="bp3-button-text"
                >
                  back
                </span>
              </button>
              <button
                class="bp3-button Button--button StyledProps--font StyledProps--main StyledProps--intent-primary Button--with-current-color Button--withRightIcon"
                type="submit"
              >
                <span
                  class="bp3-button-text"
                >
                  continue
                </span>
                <span
                  class="bp3-icon bp3-icon-chevron-right StyledProps--main"
                  icon="chevron-right"
                >
                  <svg
                    data-icon="chevron-right"
                    height="16"
                    viewBox="0 0 16 16"
                    width="16"
                  >
                    <desc>
                      chevron-right
                    </desc>
                    <path
                      d="M10.71 7.29l-4-4a1.003 1.003 0 00-1.42 1.42L8.59 8 5.3 11.29c-.19.18-.3.43-.3.71a1.003 1.003 0 001.71.71l4-4c.18-.18.29-.43.29-.71 0-.28-.11-.53-.29-.71z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
